textattack/transformations/word_swap_inflections.py

```diff
@@ -0,0 +1,89 @@
+from flair.data import Sentence
+from flair.models import SequenceTagger
+import lemminflect
+
+from textattack.transformations.word_swap import WordSwap
+
+
+class WordSwapInflections(WordSwap):
+    """Transforms an input by replacing its words with their inflections.
+
+    For example, the inflections of 'schedule' are {'schedule', 'schedules',
+    'scheduling'}.
+
+    Base on ``It’s Morphin’ Time! Combating Linguistic Discrimination with
+    Inflectional Perturbations".
+    (https://www.aclweb.org/anthology/2020.acl-main.263.pdf)
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self._flair_pos_tagger = SequenceTagger.load("pos-fast")
+        self._flair_to_lemminflect_pos_map = {"NN": "NOUN", "VB": "VERB", "JJ": "ADJ"}
+
+    def _get_replacement_words(self, word, word_part_of_speech):
+        if word_part_of_speech not in self._flair_to_lemminflect_pos_map:
+            # Only nouns, verbs, and adjectives have proper inflections.
+            return []
+        replacement_inflections_dict = lemminflect.getAllLemmas(word)
+        # `lemminflect.getAllLemmas` returns a dict mapping part-of-speech
+        # to available inflections. First, map part-of-speech from flair
+        # POS tag to lemminflect.
+        lemminflect_pos = self._flair_to_lemminflect_pos_map[word_part_of_speech]
+        return replacement_inflections_dict.get(lemminflect_pos, None)
+
+    def _get_transformations(self, current_text, indices_to_modify):
+        words = current_text.words
+        sentence = Sentence(" ".join(words))
```

Comment:
why can't you pass in `current_text.text`?

Reply:
you can, this is better